src/OpenTelemetry.Exporter.Jaeger/JaegerExporter.cs

```diff
@@ -0,0 +1,128 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+namespace OpenTelemetry.Exporter.Jaeger
+{
+    using System;
+    using OpenTelemetry.Exporter.Jaeger.Implimentation;
+    using OpenTelemetry.Trace.Export;
+
+    public class JaegerExporter : IDisposable
+    {
+        public const string DefaultAgentUdpHost = "localhost";
+        public const int DefaultAgentUdpCompactPort = 6831;
+        public const int DefaultMaxPacketSize = 65000;
+
+        private const string ExporterName = "JaegerTraceExporter";
+
+        private readonly object @lock = new object();
+        private readonly JaegerExporterOptions options;
+        private readonly IExportComponent exportComponent;
+
+        private bool isInitialized = false;
+        private JaegerTraceExporterHandler handler;
+        private bool disposedValue = false; // To detect redundant dispose calls
+
+        public JaegerExporter(JaegerExporterOptions options, IExportComponent exportComponent)
+        {
+            this.ValidateOptions(options);
+            this.InitializeOptions(options);
+
+            this.options = options;
+            this.exportComponent = exportComponent;
+        }
+
+        public void Start()
+        {
+            lock (this.@lock)
+            {
+                if (this.isInitialized)
+                {
+                    return;
+                }
+
+                if (this.exportComponent != null)
+                {
+                    this.handler = new JaegerTraceExporterHandler(this.options);
+                    this.exportComponent.SpanExporter.RegisterHandler(ExporterName, this.handler);
+                }
+            }
+        }
+
+        public void Stop()
+        {
+            if (!this.isInitialized)
+            {
+                return;
+            }
+
+            lock (this.@lock)
+            {
+                if (this.exportComponent != null)
+                {
+                    this.exportComponent.SpanExporter.UnregisterHandler(ExporterName);
+                }
+            }
+
+            this.isInitialized = false;
+        }
+
+        public void Dispose()
+        {
+            // Do not change this code. Put cleanup code in Dispose(bool disposing).
+            this.Dispose(true);
+        }
+
+        protected virtual void Dispose(bool disposing)
+        {
+            if (!this.disposedValue)
+            {
+                if (disposing)
+                {
+                    this.handler.Dispose();
+                }
+
+                this.disposedValue = true;
+            }
+        }
+
+        private void ValidateOptions(JaegerExporterOptions options)
+        {
+            if (string.IsNullOrWhiteSpace(options.ServiceName))
+            {
+                throw new ArgumentNullException("ServiceName", "Service Name is required.");
```

Comment:
nit: Perhaps it should be simply `ArgumentException` where the property `ServiceName` of the argument `options` was expected?
`ArgumentNullException` I believe exist to indicate the parameter was actually null.